### ISSUE TITLE
Issue when setting gid (needs different group id for my User)

### ISSUE THREAD

**Issue by stivesso** [deric/puppet-accounts#42]
Hi,

I'm willing to create some users with different GID, but when I'm trying to do so, I then get error during puppet run , it seems for me that the fact that `primary_group` is used to set home user permission (when `managehome` is true) is the main Issue from what I am trying to achieve, see below:

**Hiera**
```yaml
accounts::users:
  test:
    gid: "1003"
    manage_group: false
    comment: "Test User"
    groups: ["med"]
    shell: "/bin/bash"
    pwhash: "$6$GDH43O5m$FaJsdjUta1wXcITgKekNGUIfrqxYogW"
    managehome: true
```
**Error**
```
Warning: Local environment: "production" doesn't match server specified node environment "dev", switching agent to "dev".
Info: Retrieving pluginfacts
Info: Retrieving plugin
Info: Loading facts
Info: Caching catalog for xxxxxxxxxxxxxxxx
Info: Applying configuration version '1458991579'
Error: Could not set 'directory' on ensure: Could not find group test at 156:/etc/puppetlabs/code/environments/dev/modules/accounts/manifests/user.pp
Error: Could not set 'directory' on ensure: Could not find group test at 156:/etc/puppetlabs/code/environments/dev/modules/accounts/manifests/user.pp
Wrapped exception:
**Could not find group test**
Error: /Stage[main]/Accounts::Users/Accounts::User[test]/File[/home/test]/ensure: change from absent to directory failed: Could not set 'directory' on ensure: Could not find group test at 156:/etc/puppetlabs/code/environments/dev/modules/accounts/manifests/user.pp
Notice: /Stage[main]/Accounts::Users/Accounts::User[test]/File[/home/test/.ssh]: Dependency File[/home/test] has failures: true
Warning: /Stage[main]/Accounts::Users/Accounts::User[test]/File[/home/test/.ssh]: Skipping because of failed dependencies
Notice: /Stage[main]/Accounts::Users/Accounts::User[test]/File[/home/test/.ssh/authorized_keys]: Dependency File[/home/test] has failures: true
Warning: /Stage[main]/Accounts::Users/Accounts::User[test]/File[/home/test/.ssh/authorized_keys]: Skipping because of failed dependencies
Notice: /Stage[main]/Accounts::Groups/Accounts::Group[med]/Group[med]: Dependency File[/home/test] has failures: true
Warning: /Stage[main]/Accounts::Groups/Accounts::Group[med]/Group[med]: Skipping because of failed dependencies
Info: Stage[main]: Unscheduling all events on Stage[main]
Notice: Applied catalog in 4.50 seconds
```

**Comment by stivesso** [comment]
I think I've a workaround for the mentioned Issue, I've created a Pull request (#43) for that.

**Comment by deric** [comment]
Resolved in #43, it will be included in next release v1.3.3.
